cfme/tests/cloud_infra_common/test_provisioning.py

```diff
@@ -378,38 +390,57 @@ def test_cloud_provision_from_template_with_attached_disks(
     if provider.one_of(AzureProvider):
         recursive_update(inst_args, {'environment': {'availability_zone': provisioning("av_set")}})
 
-    device_name = "/dev/sd{}"
+    device_name = "vd{}"
     device_mapping = []
 
-    with provider.mgmt.with_volumes(1, n=disks) as volumes:
-        for i, volume in enumerate(volumes):
-            device_mapping.append((volume, device_name.format(chr(ord("b") + i))))
-        # Set up automate
+    volumes = provider.mgmt.volume_configurations(1, n=disks)
+
+    @request.addfinalizer
+    def delete_volumes():
+        for volume in volumes:
+            provider.mgmt.delete_volume(volume)
 
-        method = modified_request_class.methods.instantiate(name="openstack_PreProvision")
+    # Set up automate
+    for i, volume in enumerate(volumes, 0):
+        if i == 0:
+            provider.mgmt.capi.volumes.set_bootable(volume, True)
+            device_mapping.append(
+                {'boot_index': 0,
+                'uuid': volume,
+                'device_name': device_name.format(chr(ord("a") + i))})
+        else:
+            device_mapping.append(
+                {'boot_index': -1,
+                'uuid': volume,
+                'device_name': device_name.format(chr(ord("a") + i))})
 
+    method = modified_request_class.methods.instantiate(name="openstack_PreProvision")
```

Comment:
Q: Here is it okay to use the hardcoded string? `"openstack_PreProvision"`?

Reply:
I think these are some names of methods that come from CFME and it is required that they are copied to other domain than they are in by default. So we cannot randomize.